source/rafcon/gui/helpers/meta_data.py

```diff
@@ -601,104 +601,110 @@ def scale_meta_data_according_states(models_dict):
 
 
 def scale_meta_data_according_state(models_dict, rel_pos=None, as_template=False, fill_up=False):
-    # TODO Documentation needed....
-    """ Scale elements in dict to fit into dict state key element.
-
-    If elements are already small enough no resize is performed.
+    """
+    Scales all elements of the dict to fit into the dict "state"-key element.
+    If the elements are small enough already, and fill_up is false, no resize is performed.
+    If rel_pos is None, or not a valid pos, the elements are positioned to
+    a valid position inside of the "state"-key element.
+
+    :param models_dict: The objects in question as dict.
+    :param (float, float) rel_pos: A position in item coordinates, relative to the parent.
+    :param bool as_template: Not used by the method.
+    :param bool fill_up: If true, the objects in question will not only be scaled, down, but also scaled up.
+    :return: The scale factor, the objects in question where scaled with, as tuple.
+    The first and second value of the tuple are always the same.
+    :rtype: (float,float)
     """
     # TODO check about positions of input-data- and output-data- or scoped variable-ports is needed
     # TODO adjustments of data ports positions are not sufficient -> origin state size is maybe needed for that
     # TODO consistency check on boundary and scale parameter for every if else case
 
+    # This tuple must always have the same value at all indices.
+    scale_factor = (1.0, 1.0)
```

Comment:
It does not make sense to have a tuple here and state in a comment that those values of the elemnts must be equal. Just use a single value then. If you lated need a tuple for compuation, just create it on the fly `(scale_factor, scale_factor)`

Reply:
I totally agree with you at this point, but since i only refactored this function, and the tuple is the return value, i decided to keep it.

Reply:
You can apply my suggested change and still return a tuple, if you do not want to refactor the calling functions!

Reply:
okay i refactor it to return a single float, and the calling functions accordingly.